### PR TITLE
Upgrade to Decidim v0.27

### PR DESCRIPTION
**Pull request by tramuntanal**
This PR upgrades the Pinto App to Decidim v0.27.1.

Release: https://github.com/decidim/decidim/releases :
- 0.27.1 : https://github.com/decidim/decidim/releases/tag/v0.27.1
- 0.27.0: https://github.com/decidim/decidim/releases/tag/v0.27.0 

Note that many Decidim configuration has moved from the `config/initializers/decidim.rb` to ENV variables. See https://github.com/decidim/decidim/pull/8725 for a full list of ENV vars.

### CODE REVIEW

File: .github/workflows/ci_app.yml [hunk 1]
```diff
@@ -30,7 +30,8 @@ jobs:
         image: redis:3.2-alpine
         ports: ["6379:6379"]
     env:
-      DB_TEST_DATABASE: pinto_app_test
+      DB_TEST_DATABASE: app-test
```
Comment (laurajaime) [inline .github/workflows/ci_app.yml:33]: Be careful!

File: .github/workflows/ci_app.yml [hunk 2]
```diff
@@ -44,12 +45,6 @@ jobs:
         with:
           ruby-version: ${{ env.RUBY_VERSION }}
           bundler-cache: true
-          
-      - uses: actions/setup-node@v2
-        with:
-          node-version: ${{ env.NODE_VERSION }}
-          cache: 'npm'
-
```
Comment (laurajaime) [inline .github/workflows/ci_app.yml:52]: Why do you remove that?

File: config/application.example.yml [hunk 3]
```diff
@@ -1,11 +1,23 @@
...
-  DB_POOL: <%= ENV.fetch("RAILS_MAX_THREADS") { "5" } %>
+  DB_POOL: "<%= ENV.fetch("RAILS_MAX_THREADS") { "5" } %>"
   DB_HOST: <%= ENV.fetch("DATABASE_HOST") { "localhost" } %>
-  DB_DATABASE: "database name"
-  DB_USERNAME: "database username"
-  DB_PASSWORD: "database password"
+  DB_DATABASE: "clean_app_dev"
+  DB_TEST_DATABASE: "clean_app_test"
+  DB_USERNAME: "db_username"
+  DB_PASSWORD: "db_username_password"
```
Comment (laurajaime) [inline config/application.example.yml:20]: can you review examples names? there are related to clean app but maybe is better use "Pinto" names

File: config/environments/production.rb [hunk 4]
```diff
@@ -76,12 +76,12 @@
     address: Rails.application.secrets.smtp_address,
     port: Rails.application.secrets.smtp_port,
     authentication: Rails.application.secrets.smtp_authentication,
-    user_name: Rails.application.secrets.smtp_authentication_username,
+    user_name: Rails.application.secrets.smtp_username,
```
Comment (laurajaime) [inline config/environments/production.rb:79]: Be careful!!!! that is needed to email works!

File: config/environments/production.rb [hunk 5]
```diff
@@ -76,12 +76,12 @@
...
     password: Rails.application.secrets.smtp_password,
     domain: Rails.application.secrets.smtp_domain,
     enable_starttls_auto: Rails.application.secrets.smtp_starttls_auto,
     openssl_verify_mode: "none",
-    ssl: false
+    ssl: true
```
Comment (laurajaime) [inline config/environments/production.rb]: I think that must be `false`.

File: config/initializers/carrierwave.rb [hunk 6]
```diff
@@ -7,25 +7,25 @@
...
-      aws_access_key_id: Rails.application.secrets.aws[:access_key_id], # required
-      aws_secret_access_key: Rails.application.secrets.aws[:secret_access_key], # required
+      aws_access_key_id: Rails.application.secrets.dig(:storage, :s3, :access_key_id), # required
+      aws_secret_access_key: Rails.application.secrets.dig(:storage, :s3, :secret_access_key), # required
       region: region, # optional, defaults to 'us-east-1'
       host: "s3.#{region}.amazonaws.com" # optional, defaults to nil
     }
-    config.fog_directory = Rails.application.secrets.aws[:bucket] # required
+    config.fog_directory = Rails.application.secrets.dig(:storage, :s3, :bucket) # required
     config.fog_public = true # (optional) public readability, defaults to true
```
Comment (laurajaime) [inline config/initializers/carrierwave.rb:29]: Remember these changes when deploy to Heroku!

File: config/secrets.yml [hunk 7]
```diff
@@ -42,7 +136,7 @@ development:
   smtp_address: <%= ENV["MAILER_SMTP_ADDRESS"] %>
   smtp_domain: <%= ENV["MAILER_SMTP_DOMAIN"] %>
   smtp_port: <%= ENV["MAILER_SMTP_PORT"] %>
-  smtp_starttls_auto: true
+  smtp_starttls_auto: false
```
Comment (laurajaime) [inline config/secrets.yml]: Be careful!

File: config/secrets.yml [hunk 8]
```diff
@@ -75,12 +169,11 @@ production:
   secret_key_base: <%= ENV["SECRET_KEY_BASE"] %>
   sendgrid: <%= !ENV["SENDGRID_USERNAME"].blank? %>
   smtp_username: <%= ENV["MAILER_SMTP_USER_NAME"] || ENV["SENDGRID_USERNAME"] %>
-  smtp_authentication_username: <%= ENV["MAILER_SMTP_AUTHENTICATION_USER_NAME"] %>
```
Comment (laurajaime) [inline config/secrets.yml:78]: 👀

File: config/secrets.yml [hunk 9]
```diff
@@ -75,12 +169,11 @@ production:
...
   smtp_password: <%= ENV["MAILER_SMTP_PASSWORD"] || ENV["SENDGRID_PASSWORD"] %>
   smtp_address: <%= ENV["MAILER_SMTP_ADDRESS"] || "smtp.sendgrid.net" %>
   smtp_domain: <%= ENV["MAILER_SMTP_DOMAIN"] || "heroku.com" %>
   smtp_port: <%= ENV["MAILER_SMTP_PORT"] || "587" %>
-  smtp_starttls_auto: true
+  smtp_starttls_auto: false
```
Comment (laurajaime) [inline config/secrets.yml]: 👀

File: db/schema.rb [hunk 10]
```diff
@@ -2,19 +2,18 @@
...
 # migrations use external dependencies or application code.
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 2022_11_21_131090) do
+ActiveRecord::Schema.define(version: 2022_12_28_102452) do
 
   # These are extensions that must be enabled in order to support this database
   enable_extension "ltree"
-  enable_extension "pg_stat_statements"
```
Comment (laurajaime) [inline db/schema.rb:17]: This is not necessary anymore?

**Review by laurajaime — CHANGES_REQUESTED** [review]
Good job!

Some comments about important configuration.

File: config/initializers/carrierwave.rb (hunk 6 above)
Comment (tramuntanal) [reply config/initializers/carrierwave.rb:29]: yes, I've already checked and it's ok

File: .github/workflows/ci_app.yml (hunk 1 above)
Comment (tramuntanal) [reply .github/workflows/ci_app.yml:33]: It's ok, this way we don't have conflicts when merging the clean-app

File: .github/workflows/ci_app.yml (hunk 2 above)
Comment (tramuntanal) [reply .github/workflows/ci_app.yml:52]: I'ts repeated at the end of the file

File: config/application.example.yml (hunk 3 above)
Comment (tramuntanal) [reply config/application.example.yml:20]: It's ok, this way we don't have conflicts when merging the clean-app

File: config/environments/production.rb (hunk 4 above)
Comment (tramuntanal) [reply config/environments/production.rb:79]: It's ok, see `secrets.yml`L171

File: config/secrets.yml (hunk 8 above)
Comment (tramuntanal) [reply config/secrets.yml:78]: It has been renamed to be the same as in the clean app

File: config/environments/production.rb (hunk 4 above)
Comment (laurajaime) [reply config/environments/production.rb:79]: Here `smtp_username` and `smtp_authentication_username` are different values.

File: db/schema.rb (hunk 10 above)
Comment (tramuntanal) [reply db/schema.rb:17]: not for the moment

**Comment by tramuntanal** [comment]
@laurajaime questions answered and changes applied!

**Review by laurajaime — APPROVED** [review]

**Pull request merged**